### ISSUE TITLE
None of app delegates are called when receiving a non onesignal remote push notification

### ISSUE THREAD

**Issue by GeraldDOW** [OneSignal/OneSignal-iOS-SDK#58]
Using release 2.0.9 still does not fix this issue. None of the following overrides are called in the app delegate:

```
-(void) application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData*)deviceToken
{
}

-(void) application:(UIApplication *)application didFailToRegisterForRemoteNotificationsWithError:(NSError *)error
{
}

-(void) application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo
{
}

- (void)application:(UIApplication *)application handleActionWithIdentifier:(nullable NSString *)identifier forRemoteNotification:(NSDictionary *)userInfo withResponseInfo:(NSDictionary *)responseInfo completionHandler:(void(^)())completionHandler
{
    
}

- (void)application:(UIApplication *)application handleActionWithIdentifier:(nullable NSString *)identifier forRemoteNotification:(NSDictionary *)userInfo completionHandler:(void(^)())completionHandler
{
    
}

- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(void (^)(UIBackgroundFetchResult result))completionHandler
{
    
}

```
Please fix this, it's a blocking issue for us.

**Comment by JKalash** [comment]
Hi,

Thank you for submitting this report. I'm trying to reproduce the bug but not being able to do so. Could you provide me with:

- Method calls to the OneSignal singleton in the initWithLaunchOptions
- Testing: device vs simulator?
- Release mode: Dev vs production?
- iOS Version / Xcode version?

**Comment by GeraldDOW** [comment]
> On Aug 17, 2016, at 5:12 PM, Joseph Kalash <notifications@github.com> wrote:
> 
> Hi,
> 
> Thank you for submitting this report. I'm trying to reproduce the bug but not being able to do so. Could you provide me with:
> 
> Method calls to the OneSignal singleton in the initWithLaunchOptions

        NSDictionary* settings = @{ kOSSettingsKeyInAppAlerts:@NO };
        
        [OneSignal initWithLaunchOptions:options appId:appID
        handleNotificationReceived:
            ^(OSNotification* notif)
            {
                // my code here
            }
         
         handleNotificationAction:
         ^(OSNotificationResult* result)
         {
             // my code here
         }
         
        settings:settings];
        
        [OneSignal IdsAvailable:^(NSString* userId, NSString* pushToken)
        {
            NSLog(@"UserId:%@", userId);
            if (pushToken != nil)
                NSLog(@"pushToken:%@", pushToken);
        }];
        
    }

> Testing: device vs simulator?

iPad 2 mini

> Release mode: Dev vs production?

Dev mode

> iOS Version / Xcode version?
iOS 9.3.4 / XCode 7.3

**Comment by JKalash** [comment]
Are you doing anything custom (non-standard) with your app delegate? 
Eg: Subclassing your AppDelegate class that conforms to UIAppDelegate?
Also, do you have any other Push Notifications framework installed?

**Comment by GeraldDOW** [comment]
> On Aug 17, 2016, at 6:04 PM, Joseph Kalash <notifications@github.com> wrote:
> 
> Are you doing anything custom (non-standard) with your app delegate? 
> Eg: Subclassing your AppDelegate class that conforms to UIAppDelegate?
> 

Yes I have TTAppController deriving from AppController implementing protocol UIApplicationDelegate

The methods are overridde in the middle class (AppController).

hum, I understand what you mean...
> Also, do you have any other Push Notifications framework installed?
> 
> 
No but we do have some custom code that handles the notifications sent by our server.

**Comment by GeraldDOW** [comment]
OK doing this works:
-(void) application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo
{
    [super application:application didReceiveRemoteNotification:userInfo];
}

because I have 3 layers of inheritance for my app delegate : B -> A -> UIApplicationDelegate

Code swizzling is definitely not a good idea when it comes to building a sdk…

**Comment by JKalash** [comment]
Thank you for providing me with this info. This is what's causing the issue. Our SDK swizzles the UIApplication class in order to save a pointer to your AppDelegate class which we expect to be potentially overriding the UIAppDelegate protocol methods you mentioned above. From what I understand, you have a subclass of your App Delegate in which you override these methods and as such we fail to see your App Delegate class overriding them and as such we
\

We will release an update to our SDK tackling the issue. For the moment being and for a quick fix, you could define these methods in the parent App Delegate and they would get called properly.

**Comment by JKalash** [comment]
Solved [2.0.10](https://github.com/OneSignal/OneSignal-iOS-SDK/releases/2.0.10)
